src/mscorlib/src/System/Globalization/CultureData.Unix.cs

```diff
@@ -14,6 +14,20 @@ namespace System.Globalization
 {
     internal partial class CultureData
     {
+        private static bool s_invariantGlobalizationMode = InitGlobalizationInvariantMode();
```

Comment:
This should be initialized explicitly during startup, not using implicit constructor. The implicit constructor as written will turn into two checks each time. 1. have we initialized already? 2. What the value is? So these checks are twice as expensive than what they need to be.

Or even better - we may want to cache the flag in a field in the objects with hot paths like TextInfo, and check the instance field there. Instance fields are faster to access than static fields.